### PR TITLE
Integration/rpnv1 742 validator selection

### PR DESCRIPTION
**Pull request by clstrfsck**
Validator selection from genesis and staked validators.

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStakeComputerImpl.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Helper class to maintain staked amounts from the radix engine state.
+ */
+@NotThreadSafe
+public final class RadixEngineStakeComputerImpl implements RadixEngineStakeComputer {
```
Comment (talekhinezh) [inline radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStakeComputerImpl.java]: Prefer a more descriptive class name like RRITokenStakeComputer

File: radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineValidatorsComputer.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package com.radixdlt.statecomputer;
+
+import com.google.common.collect.ImmutableSet;
+import com.radixdlt.crypto.ECPublicKey;
+
+/**
+ * Radix engine state computer to keep track of registered validators.
+ */
+public interface RadixEngineValidatorsComputer {
```
Comment (talekhinezh) [inline radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineValidatorsComputer.java:26]: Seems to be only one implementation of this. Would prefer the interface removed and use of just the single class.

**Review by talekhinezh — COMMENTED** [review]
Some minor comments.

One thing would like to see is a simulation test which tests validators getting kicked out across epochs (e.g. test with maxValidators=1) as new validators get more stake. Doesn't need to be this PR but atleast in the next one.

File: radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineValidatorsComputer.java (hunk 2 above)
Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineValidatorsComputer.java:26]: There is also `SimulationValidatorComputersModule.SimulationValidatorComputer` that allows preload with an initial set.

Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineValidatorsComputer.java:26]: I expect that there will be more for specific cases as we add more tests.

File: radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStakeComputerImpl.java (hunk 1 above)
Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStakeComputerImpl.java]: No probs.  Will change.

Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStakeComputerImpl.java]: I ended up with `TokenStakeComputer`.  Not really sure the use of "RRI" really adds anything.

**Pull request merged**